backend/howtheyvote/analysis/votes.py

```diff
@@ -247,3 +247,64 @@ def _match_by_position_counts(self, vote: Vote) -> set[str]:
             return set()
 
         return self._releases_by_date_position_counts[key]
+
+
+class TopicsAnalyzer:
+    def __init__(self, vote: Vote):
+        self.vote = vote
+
+    def run(self) -> Fragment | None:
+        if self.vote.oeil_subjects == []:
+            return None
+
+        topics = []
+
+        oeil_subjects = " ".join([subject.code for subject in self.vote.oeil_subjects])
+
+        subject_to_topic = {
```

Comment:
Long term it might make sense to split this up into different mappings for different topics, I find this feasible for the moment...

Reply:
I think it’s fine like this, though you could make the mapping a class variable of `TopicsAnalyzer` so that the dict isn’t recreated for every single instance.

Reply:
:ballot_box_with_check: